docs/jenkins.md

```diff
@@ -0,0 +1,194 @@
+
+An org admin must then:
+
+* add the repo to the [Bots team](https://github.com/orgs/theforeman/teams/bots/repositories) with **write** access
+
+## Job configurations
+
+### Testing develop
+
+All repos with an associated job that tests their master/develop branch should have a hook added to the repo to trigger immediate builds.
+
+To set up the hook, an org/repo admin must:
+
+* View the repository settings
+* Click _Webhooks & Services_
+  * _Add service_ dropdown, select _Jenkins (Git plugin)_
+* For the URL, enter `https://ci.theforeman.org/`
+
+### Pull request testing
+
+Core Foreman projects have GitHub pull requests tested on our Jenkins instance so it's identical to the way we test the primary development branches themselves.  Simpler and quieter projects (such as installer modules) should use Travis CI which supports PR testing and reduces the load on our own infrastructure.
+
+Every project that needs PR testing has at least two Jenkins jobs.  Taking core Foreman as an example:
+
+* Test job for the main development branch (develop or master): [test_develop](https://ci.theforeman.org/job/test_develop/)
+* Test job for each PR: [test_develop_pr_core](https://ci.theforeman.org/job/test_develop_pr_core/)
+
+#### Github Pull Request Builder
+
+The GHPRB plugin uses webhooks installed on the repo to trigger a build, then it runs any job configured with the GHPRB trigger and a matching GitHub project set.
+
+The plugin tests the latest commit on the PR branch only, it does not merge the PR with the base branch. The webhook may also trigger multiple jobs, and jobs may use different GitHub commit status names to easily test and report status on different types of tests.
+
+PR jobs should be set up identically to primary branch tests, except for the SCM (which checks out `${sha1}`) and to add the GHPRB trigger (see the `github_pr` macro in JJB).
+
+To set up the hook, an org/repo admin goes to the repository settings, then Webhooks & Services and adds a webhook with these settings:
+
+* Payload URL: `https://ci.theforeman.org/ghprbhook/`
+* Content type: `application/json`
+* Secret: _redacted_
+* Events: _Let me select individual events_, _Pull request_, _Issue comment_
+
+An org admin must then change the org teams:
+
+* Add the repository to the [Bots team](https://github.com/orgs/theforeman/teams/bots/repositories) with **write** access
+
+### CentOS QA service
+
+CentOS offer access to their [QA CI infrastructure](https://wiki.centos.org/QaWiki/CI) to Foreman and other projects, which gives on-demand access to run tests on physical hardware running CentOS.  It consists of three main components:
+
+* [ci.centos.org](https://ci.centos.org) - a Jenkins instance which we can manage jobs on
+* foreman@slave01 - a user account on a slave which our jobs all run on
+* [Duffy](https://wiki.centos.org/QaWiki/CI/Duffy) - on-demand provisioning of physical test servers running CentOS, available from foreman`slave01 jobs
```

Comment:
The slave01 part it no really correct anymore. I think we now limit this the required secret to OSUOSL nodes.

Reply:
That's the account on ci.centos.org infra. It at least that's how I read that section and it matches my knowledge of the infra.

Reply:
https://ci.centos.org/computer/foreman-ci-slave01/

Reply:
ah right